services/QuillLMS/app/controllers/teachers/classroom_manager_controller.rb

```diff
@@ -8,7 +8,7 @@ class Teachers::ClassroomManagerController < ApplicationController
 
   respond_to :json, :html
 
-  around_action :force_writer_db_role, only: [:assign, :dashboard, :lesson_planner]
+  around_action :force_writer_db_role, only: [:assign, :dashboard, :lesson_planner, :view_demo]
```

Comment:
Note, this adds a `get` link that has writes in it (to clean-up the demo account before the user gets there). We could make that fire off an async worker (that's what the leave demo account link does), but that wouldn't guarantee the demo account is in a good place when the teacher lands there(in all likelihood, it would not be reset by the time they land). Also, I believe, we want this to be a url that is clickable, shareable, rather than `post`.

This feels like we should make an exception to our "no new GET writes" policy for this one link, but open to feedback.

Reply:
I'm on board with this being an intentional exception.  We shouldn't be dogmatic about rules like that, and I think you're right about sharability.  Also, I can easily imagine Coaches or other staff keeping bookmarks for the demo, where it will again make sense to be a GET.  I feel like the alternative would be some sort of Pusher-based solution where you land on a page that starts a background job and gives you a spinner until the job completes (and sends a Pusher notification) back to the front-end client.  It's not clear to me that it's worth the extra complexity just to keep all GETs from having writes in them.